libwallet/assets/btc/dex-wallet.go

```diff
@@ -31,142 +35,291 @@ import (
 	"github.com/lightninglabs/neutrino/headerfs"
 )
 
-// DEXWallet wraps *wallet.Wallet and implements dexbtc.BTCWallet.
+// DEXWallet wraps *wallet.Wallet and implements dexbtc.Wallet.
 type DEXWallet struct {
-	w          *wallet.Wallet
-	acctNum    int32
-	spvService *btcChainService
+	w        *wallet.Wallet
+	acctNum  int32
+	cl       *btcChainService
+	syncData *SyncData
 }
 
-var _ dexbtc.BTCWallet = (*DEXWallet)(nil)
+var _ dexbtc.CustomWallet = (*DEXWallet)(nil)
 
 // NewDEXWallet returns a new *DEXWallet.
-func NewDEXWallet(w *wallet.Wallet, acctNum int32, nc *chain.NeutrinoClient) *DEXWallet {
+func NewDEXWallet(w *wallet.Wallet, acctNum int32, nc *chain.NeutrinoClient, syncData *SyncData) *DEXWallet {
 	return &DEXWallet{
 		w:       w,
 		acctNum: acctNum,
-		spvService: &btcChainService{
+		cl: &btcChainService{
 			NeutrinoClient: nc,
 		},
+		syncData: syncData,
 	}
 }
 
-func (dw *DEXWallet) PublishTransaction(tx *wire.MsgTx, label string) error {
-	return dw.w.PublishTransaction(tx, label)
+// Part of dexbtc.Wallet interface.
+func (dw *DEXWallet) Reconfigure(*asset.WalletConfig, string) (bool, error) {
+	return false, errors.New("Reconfigure not supported for Cyptopower btc wallet")
 }
 
-func (dw *DEXWallet) CalculateAccountBalances(account uint32, confirms int32) (wallet.Balances, error) {
-	return dw.w.CalculateAccountBalances(account, confirms)
+// Part of dexbtc.Wallet interface.
+func (dw *DEXWallet) RawRequest(_ context.Context, _ string, _ []json.RawMessage) (json.RawMessage, error) {
+	// Not needed for spv wallet.
+	return nil, errors.New("RawRequest not available on spv")
 }
 
-func (dw *DEXWallet) ListUnspent(minconf, maxconf int32, acctName string) ([]*btcjson.ListUnspentResult, error) {
-	return dw.w.ListUnspent(minconf, maxconf, acctName)
+// Part of dexbtc.Wallet interface.
+func (dw *DEXWallet) OwnsAddress(addr btcutil.Address) (bool, error) {
+	return dw.w.HaveAddress(addr)
 }
 
-func (dw *DEXWallet) FetchInputInfo(prevOut *wire.OutPoint) (*wire.MsgTx, *wire.TxOut, *psbt.Bip32Derivation, int64, error) {
-	return dw.w.FetchInputInfo(prevOut)
-}
+/*
```

Comment:
The notes should be inside the method implementation. It's an implementation detail.

Reply:
Errm, ok.